File: frontend/src/types/prediction.ts
// Mirrors backend/app/schemas/prediction.py
export interface PredictionRequest {
  location: string;
  area_sqft: number;
  floor_num: number;
  bathroom: number;
  balcony: number;
  car_parking: number;
  furnishing: string;
  transaction: string;
  ownership: string;
  facing: string;
  overlooking: string;
  status: string;
}

export interface PredictionResponse {
  predicted_price: number;
}

export interface ApiError {
  detail: string;
}

export interface CategoryOptions {
  furnishing: string[];
  transaction: string[];
  ownership: string[];
  facing: string[];
  overlooking: string[];
  status: string[];
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel, Field


class PredictionRequest(BaseModel):
    location: str
    area_sqft: float = Field(gt=0, description="Property area in square feet")
    floor_num: int
    bathroom: int = Field(ge=0)
    balcony: int = Field(ge=0)
    car_parking: int = Field(ge=0)
    furnishing: str  # "Furnished" | "Semi-Furnished" | "Unfurnished"
    transaction: str  # "New Property" | "Resale"
    ownership: str
    facing: str
    overlooking: str
    status: str


class PredictionResponse(BaseModel):
    predicted_price: float